deserializers/src/main/java/com/nextdoor/bender/deserializer/json/GenericJsonEvent.java

```diff
@@ -122,13 +122,17 @@ public Object removeField(String fieldName) throws FieldNotFoundException {
       throw new FieldNotFoundException(fieldName + " is not in payload because payload is null");
     }
 
-    JsonObject json = this.payload.getAsJsonObject();
-    JsonElement elm = JsonPathProvider.read(json, fieldName);
```

Comment:
I don't even understand how this used to work?

Reply:
I think the way the diff is visualized it makes it look odd. 

It used to read
```
JsonObject json = this.payload.getAsJsonObject();
JsonElement elm = JsonPathProvider.read(json, fieldName);
JsonPathProvider.delete(json, fieldName);
return elm;
```

but now reads
```
Object o = getField(fieldName);
JsonPathProvider.delete(this.payload, fieldName);
return o;
```